yocto-base/00_flash_and_verify.robot

```diff
@@ -0,0 +1,49 @@
+*** Settings ***
+Library             Collections
+Library             String
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+Resource            ../../sonoff-rest-api/sonoff-api.robot
+Resource            ../../rtectrl-rest-api/rtectrl.robot
+Resource            ../../variables.robot
+Resource            ../../keywords.robot
+Resource            ../lib/sd-wire.robot
+
+Suite Setup         Prepare Test Suite
+Suite Teardown      Log Out And Close Connection
+
+
+*** Test Cases ***
+Verify number of connected SD Wire devices
+    [Documentation]    Test confirms the number of connected SD Wire devices to
+    ...    the DUT platform.
+    ${sd_wire_id_list}=    Get List Of SD Wire Ids
+    ${length_of_sd_wire_id_list}=    Get Length    ${sd_wire_id_list}
+    # this is done to check whether the id of the SD Wire that's connected
+    # to the RTE matches the expected value.
+    Should Be Equal    ${length_of_sd_wire_id_list}    ${SD_WIRES_CONNECTED}
+    FOR    ${index}    IN RANGE    0    ${SD_WIRES_CONNECTED}
+        Should Be Equal    ${sd_wire_id_list[${index}]}    ${SD_WIRE_SERIAL1}
+    END
+
+Flash platform and verify
+    [Documentation]    This test flashes the DUT connected to the RTE through
+    ...    the SD Wire, then attempts to log into it over serial to see whether
+    ...    it works.
+    IF    '${PREV_TEST_STATUS}' == 'FAIL'
+        Fail    'Incorrect number of connected SD Wire devices.'
+    END
+    Variable Should Exist    ${DUT_PASSWORD}
+    # flashing
+    Flash SD Card Via SD Wire    ${FILE_BMAP}    ${FILE_GZ}    ${SD_WIRE_SERIAL1}
+    # telnet
+    Telnet.Set Prompt    :~#
+    Telnet.Read Until    login:
+    Telnet.Write Bare    \n
+    Telnet.Login    root    ${DUT_PASSWORD}
```

Comment:
There are existing keywords for logging in to Linux. It is very common operation. Perhaps these can be reused.

Reply:
@macpijan same as above, adressed [here](https://github.com/Dasharo/open-source-firmware-validation/pull/22/commits/299c9c0ba1181a5870c8002d6b1b5dd11a1c69ea#diff-322efff20647f11c797d3f64ba9852d27c8a23969970ad17b19b7b9e7c7d7029L39-R39)

Reply:
It would be ideal to use Login to Linux, but I see it is not generic enough. Please create issue in this repo linking to this comment that we should use the generic keyword in this test. Then you can close the discussion.

https://github.com/Dasharo/open-source-firmware-validation/blob/main/keywords.robot#L93